src/app/content/practiceQuestions/components/Question.tsx

```diff
@@ -40,25 +36,56 @@ const getChoiceLetter = (value: number) => {
 };
 
 // tslint:disable-next-line: variable-name
-const Question = (props: QuestionProps) => {
+const Question = () => {
   const [selectedAnswer, setSelectedAnswer] = React.useState<PracticeAnswer | null>(null);
+  const [showCorrect, setShowCorrect] = React.useState(false);
+  const question = useSelector(pqSelectors.question);
+  const section = useSelector(pqSelectors.selectedSection);
+  const questionsAndAnswers = useSelector(pqSelectors.questionsAndAnswers);
+  const dispatch = useDispatch();
+
+  React.useEffect(() => {
+    setSelectedAnswer(null);
+    setShowCorrect(false);
```

Comment:
I think in theory it could be possible. Changing this to
```js
return () => {
    setSelectedAnswer(null);
    setShowCorrect(false);
}
```
should do the job, right?
It should update the state before processing the new question?

Reply:
state updates are async while redux updates are synchronous so i'm not sure if there might still be a small race condition, but it'd be good to at least remove the `useEffect` from the equation because that will also cause additional delay